reference/preconditioner/batch_block_jacobi.hpp

```diff
@@ -0,0 +1,136 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_REFERENCE_PRECONDITIONER_BATCH_BLOCK_JACOBI_HPP_
+#define GKO_REFERENCE_PRECONDITIONER_BATCH_BLOCK_JACOBI_HPP_
+
+
+#include <ginkgo/core/preconditioner/batch_jacobi.hpp>
+
+
+#include "core/base/batch_struct.hpp"
+#include "core/matrix/batch_struct.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace host {
+
+
+/**
+ * BatchBlockJacobi preconditioner for batch solvers.
+ */
+template <typename ValueType, typename IndexType>
+class BatchBlockJacobi final {
+private:
+    inline void common_generate_for_all_system_matrix_types(size_type batch_id)
```

Comment:
that name is very cryptic, I can't tell what this is supposed to do (even after looking at the implementation)